firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLitePersistence.java

```diff
@@ -199,7 +196,40 @@ void runTransaction(String action, Runnable operation) {
   }
 
   long getByteSize() {
-    return databasePath.length();
+    return getPageCount() * getPageSize();
+  }
+
+  /**
+   * Gets the page size of the database. Typically 4096.
+   *
+   * @see https://www.sqlite.org/pragma.html#pragma_page_size
+   */
+  private long getPageSize() {
+    return pragma("page_size");
+  }
+
+  /**
+   * Gets the number of pages in the database file. Multiplying this with the page size yields the
+   * approximate size of the database on disk (including the WAL, if relevant).
+   *
+   * @see https://www.sqlite.org/pragma.html#pragma_page_count.
+   */
+  private long getPageCount() {
+    return pragma("page_count");
+  }
+
+  /**
+   * @param pragmaName Must be a valid 'pragma-name' and must refer to a long value.
+   * @see https://www.sqlite.org/pragma.html
+   */
+  private long pragma(String pragmaName) {
```

Comment:
Not all pragmas resolve to a single long value. Maybe name this simpleQueryForLong?

I suggest that name because for queries that resolve to a single 1x1 result, SQLite supports running them with `SQLiteStatement.simpleQueryForLong` in a way that avoids allocating a cursor.

Reply:
No longer relevant.